### PR TITLE
Run more unittests without requiring tensorflow

### PR DESCRIPTION
**Pull request by stweil**

### CODE REVIEW

File: unittest/baseapi_thread_test.cc [hunk 1]
```diff
@@ -23,7 +23,9 @@
 #include <functional>
 #include <memory>
 #include <string>
+#ifdef INCLUDE_TENSORFLOW
```
Comment (egorpugin) [inline unittest/baseapi_thread_test.cc:26]: Is it used only for thread pool in this test?
If yes, we should rewrite it to c++11 or abseil.

Comment (egorpugin) [reply unittest/baseapi_thread_test.cc:26]: https://github.com/abseil/abseil-cpp/blob/master/absl/synchronization/internal/thread_pool.h

Comment (stweil) [reply unittest/baseapi_thread_test.cc:26]: Yes. I'd prefer c++11 and reduce the use of abseil (which causes lots of compiler warnings).

**Comment by egorpugin** [comment]
New tests work fine for me (57/60).

**Comment by egorpugin** [comment]
@stweil 
If you remember something to change/update, put a note here (in progress col):
https://github.com/tesseract-ocr/tesseract/projects/1

I've added some general items as well as this test upgrade.

**Pull request merged**